modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+						),
+					),
+				),
+			),
+		)
+	);
+}
+add_action( 'init', 'plsr_register_setting' );
+
+/**
+ * Adds the settings sections and fields for the Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_add_setting_ui() {
+	add_settings_section(
+		'plsr_speculation_rules',
+		__( 'Speculation Rules', 'performance-lab' ),
+		static function () {
+			?>
+			<p class="description">
+				<?php esc_html_e( 'This section allows you to control how URLs that your users navigate to are speculatively loaded to improve performance.', 'performance-lab' ); ?>
+			</p>
+			<?php
+		},
+		'reading'
+	);
+
+	$fields = array(
+		'mode'      => array(
+			'title'       => __( 'Speculation Mode', 'performance-lab' ),
+			'description' => __( 'Prerendering will lead to faster load times than prefetching. However, in case of interactive content, prefetching may be a safer choice.', 'performance-lab' ),
+		),
+		'eagerness' => array(
+			'title'       => __( 'Eagerness', 'performance-lab' ),
+			'description' => __( 'The eagerness setting defines the heuristics based on which the loading is triggered.', 'performance-lab' )
+				. '<br>' . __( '"Eager" will have the minimum delay to start speculative loads, "Conservative" increases the chance that only URLs the user actually navigates to are loaded.', 'performance-lab' ),
+		),
+	);
+	foreach ( $fields as $slug => $args ) {
+		add_settings_field(
+			"plsr_speculation_rules_{$slug}",
+			$args['title'],
+			'plsr_render_settings_field',
+			'reading',
+			'plsr_speculation_rules',
+			array_merge(
+				array( 'field' => $slug ),
+				$args
+			)
+		);
+	}
+}
+add_action( 'load-options-reading.php', 'plsr_add_setting_ui' );
+
+/**
+ * Renders a settings field for the Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param array $args Associative array with 'field', 'title', and optional 'description' keys.
+ */
+function plsr_render_settings_field( array $args ) {
+	if ( empty( $args['field'] ) || empty( $args['title'] ) ) { // Invalid.
+		return;
+	}
+
+	$option = get_option( 'plsr_speculation_rules' );
+	if ( ! isset( $option[ $args['field'] ] ) ) { // Invalid.
+		return;
+	}
+
+	$value   = $option[ $args['field'] ];
+	$choices = call_user_func( "plsr_get_{$args['field']}_labels" );
+
+	?>
+	<fieldset>
+		<legend class="screen-reader-text"><?php echo esc_html( $args['title'] ); ?></legend>
+		<?php
+		foreach ( $choices as $slug => $label ) {
+			?>
+			<p>
+				<label>
+					<input
+						name="<?php echo esc_attr( "plsr_speculation_rules[{$args['field']}]" ); ?>"
+						type="radio"
+						value="<?php echo esc_attr( $slug ); ?>"
+						<?php checked( $value, $slug ); ?>
+					>
+					<?php echo esc_html( $label ); ?>
+				</label>
+			</p>
+			<?php
+		}
+
+		if ( ! empty( $args['description'] ) ) {
+			?>
+			<p class="description">
+				<?php echo wp_kses( $args['description'], array( 'br' => array() ) ); ?>
```

Comment:
If the `<br>` is removed per above, this can then be:

```suggestion
				<?php echo esc_html( $args['description'] ); ?>
```